counter.rb

```diff
@@ -0,0 +1,28 @@
+class Counter
+  def initialize(filename)
+    @filename = filename
+  end
+
+  # Do everything
+  def run
+    contents = File.read(@filename)
+    word_hash = split_text(contents)
+    print_descending(word_hash)
+  end
+
+  # Splits the text into a hash of words and counts
+  def split_text(text)
+    all_words = text.scan(/((?:\w+|\')+)/).flatten.map { |x| x.downcase }
+
+    mapping = Hash.new
```

Comment:
Ideally, I would do a `map`, followed with a `to_h`, but evidently Ruby 2.0.0 does not have `Array#to_h`.

I can try to do a similar thing with `.tap` or `.reduce`